src/Analyzers/Analyzers/Serializability/ViewModelSerializabilityAnalyzer.cs

```diff
@@ -74,65 +79,169 @@ private static void AnalyzeSyntaxTree(SemanticModelAnalysisContext context)
-            return testType.IsSerializationSupported(semanticModel.Compilation, out _);
+            var testType = context.SemanticModel.Compilation.GetSpecialType(SpecialType.System_String);
+            return testType.IsKnownSerializableType(context.SemanticModel.Compilation);
         }
 
         /// <summary>
         /// Notify user about unserializable properties and unsupported properties
         /// </summary>
         /// <param name="viewModel">ViewModel class declaration</param>
         /// <param name="context">Semantic context</param>
-        private static void AnalyzeViewModelProperties(ClassDeclarationSyntax viewModel, INamedTypeSymbol? bindAttribute, INamedTypeSymbol? jsonIgnoreAttribute, SemanticModelAnalysisContext context)
+        private static void AnalyzeViewModelProperties(ClassDeclarationSyntax viewModel, SerializabilityAnalysisContext context)
         {
             var semanticModel = context.SemanticModel;
-            foreach (var property in viewModel.DescendantNodes().OfType<PropertyDeclarationSyntax>()
-                .Where(f => f.Modifiers.Any(m => m.Kind() == SyntaxKind.PublicKeyword)))
+            foreach (var property in viewModel.ChildNodes().OfType<PropertyDeclarationSyntax>())
             {
                 if (semanticModel.GetDeclaredSymbol(property) is not IPropertySymbol propertySymbol)
                     continue;
-                if (semanticModel.GetSymbolInfo(property.Type).Symbol is not ITypeSymbol propertyTypeSymbol)
-                    continue;
 
-                // Determine whether this property participates in serialization
-                if (IsSerializationIgnored(propertySymbol, bindAttribute, jsonIgnoreAttribute))
-                    continue;
+                AnalyzeProperty(propertySymbol, property.GetLocation(), "this", context);
+            }
+        }
+
+        private static void AnalyzeProperty(IPropertySymbol propertySymbol, Location location, string path, SerializabilityAnalysisContext context)
+        {
+            // Static and non-public properties do not participate in serialization
+            if (propertySymbol.IsStatic || propertySymbol.DeclaredAccessibility != Accessibility.Public)
+                return;
+
+            // Serialization might be ignored using attributes
+            if (IsSerializationIgnored(propertySymbol, context))
+                return;
+
+            AnalyzePropertyType(propertySymbol.Type, location, $"{path}.{propertySymbol.Name}", context);
+        }
+
+        private static void AnalyzePropertyType(ITypeSymbol propertyType, Location location, string path, SerializabilityAnalysisContext context)
+        {
+            // Check if we previously resolved this type
+            if (context.IsVisited(propertyType))
+            {
+                ReportIfNotSerializable(propertyType, location, path, context);
+                return;
+            }
+            context.MarkAsVisited(propertyType);
```

Comment:
Would be nice ignore these checks if the property or class has [JsonConverter] attribute.